src/main/haskell/language-kore/test/parser/Data/Kore/Step/StepTest.hs

```diff
@@ -0,0 +1,583 @@
+        ApplicationPattern Application
+            { applicationSymbolOrAlias = sigmaSymbol
+            , applicationChildren = [asAst p1, asAst p2]
+            }
+metaSigma
+    :: (MetaPattern PatternSort p1, MetaPattern PatternSort p2)
+    => p1 -> p2 -> MetaSigma p1 p2
+metaSigma = MetaSigma
+
+
+fSymbol :: SymbolOrAlias Meta
+fSymbol = SymbolOrAlias
+    { symbolOrAliasConstructor = Id "#f" AstLocationTest
+    , symbolOrAliasParams = []
+    }
+
+newtype MetaF p1 = MetaF p1
+instance (MetaPattern PatternSort p1)
+    => ProperPattern Meta PatternSort (MetaF p1)
+  where
+    asProperPattern (MetaF p1) =
+        ApplicationPattern Application
+            { applicationSymbolOrAlias = fSymbol
+            , applicationChildren = [asAst p1]
+            }
+metaF
+    :: (MetaPattern PatternSort p1)
+    => p1 -> MetaF p1
+metaF = MetaF
+
+
+gSymbol :: SymbolOrAlias Meta
+gSymbol = SymbolOrAlias
+    { symbolOrAliasConstructor = Id "#g" AstLocationTest
+    , symbolOrAliasParams = []
+    }
+
+newtype MetaG p1 = MetaG p1
+instance (MetaPattern PatternSort p1)
+    => ProperPattern Meta PatternSort (MetaG p1)
+  where
+    asProperPattern (MetaG p1) =
+        ApplicationPattern Application
+            { applicationSymbolOrAlias = gSymbol
+            , applicationChildren = [asAst p1]
+            }
+metaG
+    :: (MetaPattern PatternSort p1)
+    => p1 -> MetaG p1
+metaG = MetaG
+
+
+hSymbol :: SymbolOrAlias Meta
+hSymbol = SymbolOrAlias
+    { symbolOrAliasConstructor = Id "#h" AstLocationTest
+    , symbolOrAliasParams = []
+    }
+
+newtype MetaH p1 = MetaH p1
+instance (MetaPattern PatternSort p1)
+    => ProperPattern Meta PatternSort (MetaH p1)
+  where
+    asProperPattern (MetaH p1) =
+        ApplicationPattern Application
+            { applicationSymbolOrAlias = hSymbol
+            , applicationChildren = [asAst p1]
+            }
+metaH
+    :: (MetaPattern PatternSort p1)
+    => p1 -> MetaH p1
+metaH = MetaH
+
+asPureMetaPattern
+    :: ProperPattern Meta sort patt => patt -> CommonMetaPattern
+asPureMetaPattern patt = patternKoreToPure Meta (asAst patt)
+
+variableRenaming
+    :: MetaSort sort
+    => MetaVariable sort -> MetaVariable sort -> VariableRenaming Meta
+variableRenaming from to =
+    VariableRenaming
+        { variableRenamingOriginal = AxiomVariable (asMetaVariable from)
+        , variableRenamingRenamed =
+            ConfigurationVariable (asMetaVariable to)
+        }
+
+proposition_5_24_3
+    :: (MetaSort sort1, ProperPattern Meta sort2 patt)
+    => [FunctionalProof Meta Variable]
+    -> MetaVariable sort1
+    -> patt
+    -> UnificationProof Meta Variable
+proposition_5_24_3 functionalProof variable patt =
+    Proposition_5_24_3
+        functionalProof
+        (asMetaVariable variable)
+        (asPureMetaPattern patt)
+
+functionalVariable
+    :: MetaSort sort => MetaVariable sort -> FunctionalProof Meta Variable
+functionalVariable = FunctionalVariable . asMetaVariable
+
+runStep
```

Comment:
I'm wondering whether the next rule should be in the public module for the Stepper instead of the testing module, but I'm not sure.  What do you think about it?

Reply:
I wasn't sure either. I was waiting for something that would help me decide whether I should expose an interface that takes an int and returns another one instead of returning an IntCounter